cloudinit/distros/freebsd.py

```diff
@@ -192,5 +194,40 @@ def update_package_sources(self):
             freq=PER_INSTANCE,
         )
 
+    @lru_cache()
```

Comment:
When working with testing functions with lru_cache you may run into intermittent problems with unittests receiving the cached response for that function and not a parametrized response based on the unittest setup. You may have to call  the clear_cache method on that function. [See the following example](https://github.com/canonical/cloud-init/blob/main/tests/unittests/net/test_init.py#L1640-L1646)

Reply:
honestly, it seems that adding lru cache helped fix the tests, so I'm completely at a loss at what's going on